src/front/app/features/challenge/challenge.reducer.js

```diff
@@ -12,8 +14,12 @@ const reducer = (state = initialState, action = {}) => {
         return upsertChallenge(action.challenge, state);
     case RESET_STATE:
         return initialState;
-    case GET_USER:
+    case USER_RECEIVED:
         return upsertCurrentUser(action.user, state);
+    case ACCEPTED_RECEIVED:
+        return state.setIn(['currentUser', 'acceptedChallenges'], fromJS(action.challenges));
+    case WATCHLIST_RECEIVED:
+        return state.setIn(['currentUser', 'watchList'], fromJS(action.challenges));
     case ADDED_TO_WATCHLIST:
         return state.set('isWatched', true);
```

Comment:
Could we resolve isWatched and isAccepted from back-end by challenge/:id endpoint?
and on accept or add to watch list just make request to back and rewrite state by responce